upgrade-2.x.sh

```diff
@@ -998,7 +1004,7 @@ if version_gt "$VERSION_ID" "2.0.6" &&
         mkdir -p $tools_path
         export PATH=$tools_path:$PATH
         download_url=https://raw.githubusercontent.com/resin-os/meta-resin/v2.3.0/meta-resin-common/recipes-support/resin-device-progress/resin-device-progress/resin-device-progress
-        curl --retry 10 -f -s -L -o $tools_path/resin-device-progress $download_url || log WARN "Couldn't download tool from $download_url, progress bar won't work, but not aborting..."
+        CURL_CA_BUNDLE=${TMPCRT} curl --retry 10 -f -s -L -o $tools_path/resin-device-progress $download_url || log WARN "Couldn't download tool from $download_url, progress bar won't work, but not aborting..."
```

Comment:
There are a couple of places in the existing code where we're hardcoding the output directory -- [/tmp/upgrade_tools](https://github.com/balena-os/resinhup/blob/master/upgrade-2.x.sh#L956) and [/tmp/upgrade_tools_extra](https://github.com/balena-os/resinhup/blob/master/upgrade-2.x.sh#L997).  It would be worth using `mktemp`, as we do up above for the curl CA bundle.

Reply:
A fair point indeed, again I'd rather change this on it's own. These dirs predate my involvement with this process, but I suspect they aren't mktemps to make future debugging easier to reason about in the case that they failed, but just speculation.